File: p050/p50.php
<?php

/**
Consecutive prime sum

The prime 41, can be written as the sum of six consecutive primes:

	41 = 2 + 3 + 5 + 7 + 11 + 13

This i sthe longest sum of consecutive primes that adds to a prime below one-hundred.

The longest sum of conescutive primes below one-thousand that adds to a prime, contains 21 terms, and
is equal to 953.

Which prime, below one-million, can be written as the sum of the most consecutive primes?
**/

require_once "../lib/prime.php";

$limit = 1000000;
$sieve = prime_sieve($limit);
$sievebound = count($sieve);

$maxlen = 0;
$max = -1;

for($i = $sievebound - 1; $i >= $sievebound - 10000; $i--) {
	$prime = $sieve[$i];
	for($j = $i - 1; $j >= 0; $j--) {
    	$sum = 0;
		$len = 0;
		for($k = $j; $k >= 0; $k--) {
			$next = $sieve[$k];
			$sum += $next;
			$len++;
			if($sum == $prime) {
				if($len > $maxlen) {
					$maxlen = $len;
					$max = $prime;
				}
			} else if($sum > $prime) {
				break;
			}
		}
	}
}

print $max;

?>

File: p050/p50.c
#include <stdio.h>

#include "../lib/prime.h"

int main(int argc, char* argv[]) {

    int max_len = 0;
    int max = -1;

    int limit = 1000000;
    sieve_t sieve = prime_sieve(limit);

    for(int i = sieve.num - 1; i > sieve.num - 10000; i--) {
        int prime = sieve.primes[i];
        for(int j = i - 1; j >= 0; j--) {
            int sum = 0;
            int len = 0;
            for(int k = j; k >= 0; k--) {
                sum += sieve.primes[k];
                len++;
                if(sum == prime) {
                    if(len > max_len) {
                        max_len = len;
                        max = prime;
                    }
                } else if(sum > prime) {
                    break;
                }
            }
        }
    }

    printf("%i", max);

    return 0;
}

